src/Libraries/Microsoft.Extensions.TimeProvider.Testing/FakeTimeProvider.cs

```diff
@@ -175,7 +175,17 @@ public override long GetTimestamp()
 #pragma warning disable S3236 // Caller information arguments should not be provided explicitly
         Debug.Assert(TimestampFrequency == TimeSpan.TicksPerSecond, "Assuming frequency equals ticks per second");
 #pragma warning restore S3236 // Caller information arguments should not be provided explicitly
-        return _now.Ticks;
+
+        DateTimeOffset result;
+
+        lock (Waiters)
+        {
+            result = _now;
+            _now += _autoAdvanceAmount;
+        }
+
+        WakeWaiters();
+        return result.Ticks;
```

Comment:
Can something like this work:

```suggestion

        return GetUtcNow().Ticks;
```

Reply:
We can, changed